README.md

```diff
@@ -58,3 +58,79 @@ xRequestId := req.Header.Get(athenahealth.XRequestIDHeaderKey))
 ```
 
 See the athena [Best Practices](https://docs.athenahealth.com/api/guides/best-practices) guide for more details about X-Request-Id and other recommended  practices.
+
+## Method Signatures Required vs. Optional Fields
+
+All methods that perform network or filesystem IO will accept a context for idiomatic propagation.
+
+While not yet consistently applied across this repo, required fields should be surfaced as top-level arguments in method signatures (including path parameters, query parameters, and request body fields) while optional fields should be embedded as pointer fields in an optional pointer struct. The aim is to provide a consistent and natural pattern for interacting with this client interface as well as to add compile-time safety to required fields.
```

Comment:
Could we also add as a benefit that this provides clear & built-in documentation on required vs optional fields?